File: packages/mobile/src/api/auth.api.ts
import apiClient from './client';
import type { User, Tokens } from '../types/models.types';

interface LoginResponse {
  success: true;
  data: Tokens & { user: User };
}

interface RefreshResponse {
  success: true;
  data: { accessToken: string; expiresIn: number };
}

export const authApi = {
  login: async (employeeId: string, password: string) => {
    const response = await apiClient.post<LoginResponse>('/auth/login', {
      employeeId,
      password,
    });
    return response.data.data;
  },

  refresh: async (refreshToken: string) => {
    const response = await apiClient.post<RefreshResponse>(
      '/auth/refresh',
      null,
      { headers: { Authorization: `Bearer ${refreshToken}` } },
    );
    return response.data.data;
  },
};

File: packages/mobile_flutter/lib/api/auth_api.dart
import 'package:dio/dio.dart';

class AuthApi {
  final Dio dio;

  AuthApi(this.dio);

  Future<Map<String, dynamic>> login(String employeeId, String password) async {
    final response = await dio.post('/auth/login', data: {
      'employeeId': employeeId,
      'password': password,
    });
    final data = response.data['data'] as Map<String, dynamic>;
    return data;
  }

  Future<String> refresh(String refreshToken) async {
    final response = await dio.post(
      '/auth/refresh',
      options: Options(
        headers: {'Authorization': 'Bearer $refreshToken'},
      ),
    );
    return response.data['data']['accessToken'] as String;
  }
}
